test/models/exam_test.rb

```diff
@@ -4,4 +4,39 @@ class ExamTest < ActiveSupport::TestCase
   test 'exam factory builds valid exam' do
     assert build(:exam).valid?
   end
+
+  test 'exam time_window calculation' do
+    start = DateTime.now
+    exam = create(
+      :exam,
+      start_time: start,
+      end_time: start + 30.minutes,
+      duration: 29
```

Comment:
I wonder if it makes sense to redefine the `duration` field to be a proper `Duration`, so that you would write `duration: 29.minutes` here instead.  https://stackoverflow.com/a/45359178/783424 seems potentially relevant.

Reply:
this is a good idea

Reply:
We could keep an int and represent it as seconds, and then we could use `29.minutes` here.

Reply:
Ok, what do you think about fc74a2c?